tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1418,6 +1419,58 @@ public function testCommentInTable(): void
         $table = $this->schemaManager->listTableDetails('table_with_comment');
         self::assertSame('Foo with control characters \'\\', $table->getComment());
     }
+
+    public function testCreatedCompositeForeignKeyOrderIsCorrectAfterCreation(): void
+    {
+        $localColumns = ['child_col2', 'child_col1'];
+        $foreignColumns = ['col2', 'col1'];
+
+        $table = new Table('test_table_foreign_key_referenced');
+
+        $table->addOption('collate', 'utf8_general_ci');
+        $table->addOption('charset', 'utf8');
+        $table->addOption('comment', 'Test foreign key referenced table');
+
+        $table->addColumn('col1', Types::INTEGER, ['notNull' => true]);
+        $table->addColumn('col2', Types::INTEGER, ['notNull' => true]);
+        $table->setPrimaryKey($foreignColumns);
+
+        $this->schemaManager->dropAndCreateTable($table);
+
+        $table = new Table('test_table_with_foreign_key');
+
+        $table->addOption('collate', 'utf8_general_ci');
+        $table->addOption('charset', 'utf8');
+        $table->addOption('comment', 'Test table with foreign key');
+
+        $table->addColumn('id', Types::INTEGER, ['autoincrement' => true, 'notNull' => true]);
+        $table->addColumn('child_col1', Types::INTEGER, ['notNull' => true]);
+        $table->addColumn('child_col2', Types::INTEGER, ['notNull' => true]);
+        $table->setPrimaryKey(['id']);
+
+        $table->addForeignKeyConstraint(
+            'test_table_foreign_key_referenced',
+            $localColumns,
+            $foreignColumns,
+            [],
+            'fk_test_foreign_key'
+        );
+
+        $this->schemaManager->dropAndCreateTable($table);
+
+        $table = $this->schemaManager->listTableDetails('test_table_with_foreign_key');
+
+        self::assertSame(
+            $localColumns,
+            $table->getForeignKey('fk_test_foreign_key')->getLocalColumns(),
+            'Order for the foreign key local columns should match the specified order for the foreign key constraint.'
```

Comment:
Personally, I find custom assertion messages doing more harm than good. An assertion may fail not only because the order doesn't match, so this message will be misleading. I'd remove it completely.

Reply:
Done. Though phpunit would output the differences too but I am happy with either solution. Personally, I very seldom use custom error messages too, but I saw some other tests using  them, so I thought it might be expected.